File: Binary Search I/python/Day 1/binary_search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        s=0
        e=len(nums)-1
        while(s<=e):
            mid=(s+e)//2
            if nums[mid]==target:
                return mid
            if nums[mid]>target:
                e=mid-1
            else:
                s=mid+1
        return -1

File: Binary Search I/cpp/Day 1/BinarySearch.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        return binSearch(nums,0,nums.size()-1,target);
    }
    int binSearch(vector<int>& num,int s,int e,int target){
        if(s>e){
            return -1;
        }
        int mid=(s+e)/2;
        if(num[mid]==target){
            return mid;
        }
        if(num[mid]>target){
            return binSearch(num,s,mid-1,target);
        }else{
            return binSearch(num,mid+1,e,target);
        }
    }
};